src/corelibs/U2View/src/ov_msa/view_rendering/MaEditorSelection.h

```diff
@@ -67,6 +73,55 @@ class U2VIEW_EXPORT MaEditorSelection {
     QList<QRect> rectList;
 };
 
+/** MSA/MCA editor selection controller. */
+class U2VIEW_EXPORT MaEditorSelectionController : public QObject {
+    Q_OBJECT
+public:
+    explicit MaEditorSelectionController(MaEditor *editor);
+
+    /** Returns current selection state. */
+    const MaEditorSelection &getSelection() const;
+
+    /** Sets new selection instance. Emits si_selectionChanged signal. */
+    virtual void setSelection(const MaEditorSelection &selection);
+
+signals:
+
+    /** Signal emitted every time selection is changed. */
+    void si_selectionChanged(const MaEditorSelection &current, const MaEditorSelection &prev);
+
+public slots:
+    /** Sets selection to empty selection. Emits signal that selection is changed. */
+    virtual void clearSelection();
+
+protected:
+    /** MSA/MCA editor instance. Never null. */
+    MaEditor *const editor;
```

Comment:
The fact that `McaEditorSelectionController` has two pointers to an editor looks a bit strange. Maybe, it's better to make this one private?

Reply:
Good idea. Done